codex/utils/asynciter.nim

```diff
@@ -4,8 +4,8 @@ import pkg/questionable
 import pkg/chronos
 
 type
-  Function*[T, U] = proc(fut: T): U {.raises: [CatchableError], gcsafe, noSideEffect.}
-  IsFinished* = proc(): bool {.raises: [], gcsafe, noSideEffect.}
+  Function*[T, U] = proc(fut: T): U {.raises: [CatchableError], gcsafe, closure.}
```

Comment:
Closure is unnecessary here, when declaring a proc type, it's implicitely a closure.